protocol/src/common/protocol.ts

```diff
@@ -1871,17 +1871,42 @@ export interface DidChangeWatchedFilesRegistrationOptions {
 	watchers: FileSystemWatcher[];
 }
 
+export interface GlobPattern {
```

Comment:
I would name this RelativePattern and use the type GlobPattern as type GlobPattern = RelativePattern | string;

We might discover other pattern forms in the future.

Reply:
thank you for review @dbaeumer! will make the change!